include/seqan3/argument_parser/detail/version_check.hpp

```diff
@@ -0,0 +1,417 @@
+    //!\brief The message directed to the developer of the app if the app is not yet registered with us.
+    static constexpr char const * const MESSAGE_UNREGISTERED_APP =
+        "[SEQAN3 INFO] :: Thank you for using SeqAn!\n"
+        "[SEQAN3 INFO] :: Do you wish to register your app for update notifications?\n"
+        "[SEQAN3 INFO] :: Just send an email to support@seqan.de with your app name and version number.\n"
+        "[SEQAN3 INFO] :: If you don't wish to receive further notifications, set --version-check OFF.\n\n";
+    //!\brief The message directed to the developer if the application is registered but under a lower version.
+    static constexpr char const * const MESSAGE_REGISTERED_APP_UPDATE =
+        "[APP INFO] :: We noticed the app version you use is newer than the one registered with us.\n"
+        "[APP INFO] :: Please send us an email with the new version so we can correct it (support@seqan.de)\n\n";
+    //!\brief The message directed to the user of the app if a new app version is available.
+    static constexpr char const * const MESSAGE_APP_UPDATE =
+        "[APP INFO] :: A new version of this application is now available.\n"
+        "[APP INFO] :: Visit www.seqan.de for updates of official SeqAn applications.\n"
+        "[APP INFO] :: If you don't wish to receive further notifications, set --version-check OFF.\n\n";
+
+    //!\brief The url to send the server request.
+    std::string url;
+    //!\brief The application name.
+    std::string name;
+    //!\brief The version of the application.
+    std::string version{"0.0.0"};
+    //!\brief The regex to verify a valid version string.
+    std::regex version_regex{"^[[:digit:]]+\\.[[:digit:]]+\\.[[:digit:]]+$"};
+    //!\brief The program used to perform the server call via a system call (wget, curl, ftp or Invoke-WebRequest).
+    std::string program;
+    //!\brief The hard coded system command that executes the `program` with the `url`.
+    std::string command;
+    //!\brief The path to store timestamp and version files (either ~/.config/seqan or the tmp directory).
+    std::filesystem::path cookie_path = get_path();
+    //!\brief The timestamp filename.
+    std::filesystem::path timestamp_filename;
+
+    //!\brief Returns a writable path to store timestamp and version files or an empty path if none exists.
+    static std::filesystem::path get_path()
+    {
+        using namespace std::filesystem;
+
+        path tmp_path;
+
+#if defined(_WIN32)
+        tmp_path = std::string{getenv("UserProfile")};
+#else
+        tmp_path = std::string{getenv("HOME")};
+#endif
+        tmp_path /= ".config";
+        tmp_path /= "seqan";
+
+        // create directory if it does not already exist
+        std::error_code ec;
+        std::filesystem::create_directory(tmp_path, ec); // does nothing and ec is zero if path already exists.
+
+        if (static_cast<bool>(ec))
```

Comment:
is the error_code not contextually convertible to bool anyway?

Reply:
yes, it has an implicit conversion operator